lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -1605,6 +1613,79 @@ bool QuakeBridgeVisitor::VisitCallExpr(clang::CallExpr *x) {
                                          reportNegateError, isAdjoint,
                                          isControl, /*paramCount=*/3);
 
+    // See if this is a custom unitary.
+    std::string maybeUnitaryGenerator = funcName.str() + "_generator_";
+    // Extract number of targets
+    size_t targetCount = 0;
+    std::string genFuncName;
+    for (auto name : customOperationNames) {
+      if (name.first.find(maybeUnitaryGenerator) != std::string::npos) {
+        targetCount =
+            std::stoi(name.first.substr(maybeUnitaryGenerator.length()));
+        genFuncName = getCudaqKernelName(name.second);
+        break;
+      }
+    }
+    if (targetCount > 0) {
+      // Extract number of parameters
+      std::size_t paramCount = [&]() {
+        std::size_t count = 0;
+        for (auto arg : args) {
+          auto argTy = arg.getType();
+          if (isa<FloatType>(argTy))
+            count++;
+          else if (auto ptrTy = dyn_cast<cc::PointerType>(argTy)) {
+            if (isa<FloatType>(ptrTy.getElementType()))
```

Comment:
I think passing parameters by reference (or pointer) should be a hard error here and unsupported. We want these functions to be pure.

Reply:
Addressed in ec6973b042fd692236c417feacf35ba893e2b5dd.